### DISCUSSION TITLE
format --diff output is not colored

### DISCUSSION THREAD

**Discussion by ssenior45** [astral-sh/ruff#9983] [Q&A] [answered]
If I run `ruff check` on a file then the output is colored which leads me to believe that my terminal is configured to display colored output from the tool correctly.

However when I run `ruff format --diff` the output is not colored. Is this expected? If so is there anything that can be done about it?

Thanks in advance.

**Comment by MichaReiser** [comment]
Hy @ssenior45 this is expected because colored diffs isn't implemented yet but it would be a nice feature to have and shouldn't be too hard to add. I created an issue for it https://github.com/astral-sh/ruff/issues/9984 with details on how it can be added. Are you interested in contributing the feature?

**Reply by yoann9344** [reply]
For now you can pipe to  `bat` : 
```bash
ruff check --diff | bat
# In some cases you have extra lines that blurs the detection, so
script_whch_lint | bat -l diff
```

**Comment by dgollahon-plaid** [comment, marked as answer]
Should this be resolved? Looks like #9984 was implemented.
